internal/cluster/common.go

```diff
@@ -272,7 +272,7 @@ func UpdateMachineCidr(db *gorm.DB, cluster *common.Cluster, machineCidr []strin
 		previousSecondaryMachineCidr = network.GetMachineCidrById(cluster, 1)
 	}
 
-	if machineCidr[0] != previousPrimaryMachineCidr || (len(machineCidr) > 1 && machineCidr[1] != previousSecondaryMachineCidr) {
+	if (len(machineCidr) > 0 && machineCidr[0] != previousPrimaryMachineCidr) || (len(machineCidr) > 1 && machineCidr[1] != previousSecondaryMachineCidr) {
```

Comment:
Not related to your code - what is the point to open transaction and not use the transaction object ?

Reply:
see my comment above on fixing it in a separate PR